### DISCUSSION TITLE
In fastapi how can we send http request from one Fastapi application to another Fastapi application.

### DISCUSSION THREAD

**Discussion by kvalli** [fastapi/fastapi#10256] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
this code in first app

from fastapi import FastAPI
app = FastAPI()

@app.post("/ProcessData")
async def Process_Data(request: Request):
     return RedirectResponse(url="http://0:0:0:0:8100/Prediction/list_x_test", status_code=status.HTTP_301_MOVED_PERMANENTLY)


i have following code in second fast api app

from fastapi import FastAPI
app = FastAPI()

@app.post("/Prediction/ldata")
def find_predict(ldata: LData)-> Response:
    json_compatible_item_data = jsonable_encoder(model_name)
        return JSONResponse(content=json_compatible_item_data)


i would like to return json response to process_data after process_data sends redirectresponse to find_predict.how to do this
```

### Description

 In fastapi how can we send http request from one Fastapi application to another Fastapi application. 
my case is im developing two fastapi applications. I have following doubts

-  can i run both fastapi applications on same linux machine at the same time
-  i need to send http request from one fast api to another  and then after http sent,the first application need to receive json response.How to achieve this.Should we have to use Httpx library inside fastapi code or will using redirectresponse will suffice
- once http post is sent to second application ,i need to capture the json response from second application .can the second application use get/post again to forward the response to first application in the same endpoint where it received http request from first application
- if i dockerise the two applications ,can i still send http request and get response from one fast api app to another .can i test this on single linux machine
- 

### Operating System

Linux

### FastAPI Version

0.95.2

### Pydantic Version

1.10.7

### Python Version

3.10.12

**Comment by chrisK824** [comment, marked as answer]
Let's see your questions:

- can i run both fastapi applications on same linux machine at the same time
sure you can. You can either run two entirely different FastAPI applications using different ports or you can use the `mount` feature to mount all sub applications to a main one, check here for this https://fastapi.tiangolo.com/advanced/sub-applications/

- i need to send http request from one fast api to another and then after http sent,the first application need to receive json response.How to achieve this.Should we have to use Httpx library inside fastapi code or will using redirectresponse will suffice
once http post is sent to second application ,i need to capture the json response from second application .can the second application use get/post again to forward the response to first application in the same endpoint where it received http request from first application
In any case you can use an HTTP client to request any server from within your application. The simplest scenario is to request your subapplication inside the endpoint handler of the main application. Then do something with the JSON response (or not) and return it from the main application endpoint. Another option, is to use the redirect response to redirect the original request to the sub application endpoint and this application endpoint can return directly the json to the caller.

- if i dockerise the two applications ,can i still send http request and get response from one fast api app to another .can i test this on single linux machine
The answer is yes. Depending which approach you will choose from second question you will either host two fastAPI applications in different ports or one main application and another one mounted on it
